Updates for CI phase 3 (#4088)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-versions/release-notes/04-Sept-2023/ci-updates-phase3-rn.md

---
title: "Update: Improvements to dbt Cloud continuous integration"

> description: "September 2023: Improved deletion of temporary schemas"

Reviewer: ```suggestion
description: "September 2023: Improved auto-deletion for temporary schemas for all adapters, PrivateLink connections, and connection string environment variables. Manual deletion may be needed in specific cases.

> description: "September 2023: Improved deletion of temporary schemas"

Reviewer: sugg adding more context for users
```suggestion
description: "September 2023: Improved auto-deletion for temporary schemas for all adapters, PrivateLink connections, and connection string environment variables. Manual deletion may be needed in specific cases."
```

Author: from what i've been taught, i lean toward having a short description so the people aren't having to reread the same thing over and over (low value). i do want to add "automatic deletion", thanks!

sidebar_label: "Update: Improved deletion of temporary schemas"
tags: [Sept-2023, CI]
date: 2023-09-18
sidebar_position: 08
---

Temporary schemas are now being deleted (dropped) for all adapters (like Databricks), PrivateLink connections, and environment variables in connection strings. 

> dbt Labs has rearchitected how schema deletion works for [continuous integration (CI)](/docs/deploy/continuous-integration) runs. We created a new service to delete any schema with a prefix of `dbt_cloud_pr` that's been generated by a PR run.

Reviewer 2: ```suggestion
dbt Labs has rearchitected how schema deletion works for [continuous integration (CI)](/docs/deploy/continuous-integration) runs. We created a new service to delete any schema with a prefix of `dbt_cloud_pr_` that's been generated by a PR run.
```

Ah, found one more.

However, temporary schemas will not be automatically deleted if:

> - Your project overrides the `generate_schema_name` macro but it doesn't contain the prefix `dbt_cloud_pr`. 

Reviewer 2: ```suggestion
- Your project overrides the `generate_schema_name` macro but it doesn't contain the prefix `dbt_cloud_pr_`. 
```

One small suggestion here: the prefix for the schema needs to end with an underscore.

> - Your project overrides the `generate_schema_name` macro but it doesn't contain the prefix `dbt_cloud_pr_`. 

Reviewer: sugg linking to the pages so users can get info in a direct way and not search for them

```suggestion
- Your project overrides the [`generate_schema_name` macro](/docs/build/custom-schemas#how-does-dbt-generate-a-models-schema-name) but it doesn't contain the prefix `dbt_cloud_pr_`.  Refer to [Troubleshooting](/docs/deploy/ci-jobs#troubleshooting) for more details.
```

Author: good flag, thanks!
